python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -128,8 +129,10 @@ def __init__(self, provider: NodeProvider):
             elif node_kind == NODE_KIND_HEAD:
                 self.head_id = node
 
-        # Note: For typical use-cases,
-        # self.all_node_ids == self.worker_ids + [self.head_id]
+        # Note: For typical use-cases, self.all_node_ids == self.worker_ids +
+        # [self.head_id]. The difference being in the case of unmanaged nodes.
+
+        self.non_terminated_nodes_time = time.time() - start_time
```

Comment:
what does this mean?

Reply:
The time it takes to call `NodeProvider.non_terminated_nodes` -- depending on implementation, this call can be a bottleneck.
Maybe we could use a comment explaining this.

Reply:
added comment.